src/app/mysterium-vpn.js

```diff
@@ -317,9 +317,7 @@ class MysteriumVpn {
       htmlContent: this._terms.getContent()
     })
 
-    const termsAnsweredDTO = await onFirstEvent((callback) => {
-      this._communication.termsAnswered.on(callback)
-    })
+    const termsAnsweredDTO = await onFirstEvent(this._communication.termsAnswered.on.bind(this))
```

Comment:
!!?? are you sure you're binding to the right object?
```suggestion
    const termsAnsweredDTO = await onFirstEvent(this._communication.termsAnswered.on.bind(this._communication.termsAnswered))
```